submissions/nadsatt/friends-app/modules/image-service.js

```diff
@@ -0,0 +1,93 @@
+export class ImageService {
+    constructor(){
+        this.totalImagesCount = 0;
+        this.imagesLoadedCount = 0;
+        this.uniqueFemaleImagesCount = 60;
+        this.uniqueMaleImagesCount = 50;
+        this.images = [];
+
+        this.borderImagePath = 'images/user-card-images/user-card-border.png';
+        this.backImagePath = 'images/user-card-images/user-card-back-background.jpg';
+
+        this.imageAlts = {
+            front: 'user-image',
+            border: 'old-gold-border-image',
+            back: 'old-texture-image'
+        };
+        this.imageClasses = {
+            border: 'border-image',
+            front: 'background-image',
+            back: 'background-image'
+        };
+    }
+
+    get femaleImagePath(){
+        const imageNumber = this.defineImageNumber(this.uniqueFemaleImagesCount);
+        return `images/female-images/female${imageNumber}.jpeg`;
+    }
+
+    get maleImagePath(){
+        const imageNumber = this.defineImageNumber(this.uniqueMaleImagesCount);
+        return `images/male-images/male${imageNumber}.jpeg`;
+    }
+
+    defineImageNumber(max, min = 1){
+        return Math.floor(Math.random() * (max - min + 1)) + min;
+    }
+
+    loadImages(users){
+        return new Promise((resolve, reject) => {
+            users = users.map(user => {
+                user.frontImage = this.createImage(
+                    this[`${user.gender}ImagePath`],
```

Comment:
Just put this gender handlers into some collection. Something like:

```
const imagePathCreators = {
  male: ...,
  female: ...,
}
imagePathCreators[user.gender]()
```

Such pattern is common used.
No need for concatenation of property name.
Image path creators are internal logic of this class, so they will be private.